### PR TITLE
Add merge to set node

### PR DESCRIPTION
**Pull request by zhugexinxin**
**Description**
I have a problem adding marks and need to merge

**Example**
```
before
  <editor>
    <block>
      <text a={{ b: 1 }}>
        word
      </text>
    </block>
  </editor>

 after
 <editor>
    <block>
      <text a={{ b: 2, c: 3 }}>
        word
      </text>
    </block>
  </editor>
```

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)
- [ ] You've [added a changeset](https://github.com/atlassian/changesets/blob/master/docs/adding-a-changeset.md) if changing functionality. (Add one with `yarn changeset add`.)

### CODE REVIEW

**Comment by BitPhinix** [comment]
I personally don't think this should be handled by slate and can be done outside it with a little helper function if desired

**Comment by zhugexinxin** [comment]
> I personally don't think this should be handled by slate and can be done outside it with a little helper function if desired, or at least not be added to the `editor.addMark` signature

I've tried helper functions before the issue.

```
before
  <editor>
    <block>
      <text>
        hello
      </text>
      <text style={{ fontSize: 1 }}>
        word
      </text>
    </block>
  </editor>
  
after
add { backgroundColor: 'blue'}

I have to do it

  const match = Array.from(Editor.nodes(editor, {
    match: (n) => !Editor.isEditor(n) && Text.isText(n),
    mode: 'lowest',
  }));
  
  match.forEach(() => {
    ...to setNodes and merge props
  })
  
but it will generate redundant undo and the handling of texts in selection is mostly duplicated in setNodes.
```

**Comment by zhugexinxin** [comment]
image
local is success.

**Comment by dylans** [comment]
> image local is success.

Thanks. They're timing out on the CI environment so we're fine there.

I'm just about to get on a long-haul flight so I don't feel comfortable merging and releasing if I'm not available in case something goes wrong. So I'll review this as soon as possible.

**Comment by zhugexinxin** [comment]
> > image local is success.
> 
> Thanks. They're timing out on the CI environment so we're fine there.
> 
> I'm just about to get on a long-haul flight so I don't feel comfortable merging and releasing if I'm not available in case something goes wrong. So I'll review this as soon as possible.

Thank you very much.

**Comment by zhugexinxin** [comment]
> I personally don't think this should be handled by slate and can be done outside it with a little helper function if desired, or at least not be added to the `editor.addMark` signature

I thought about it, you are right, no changes should be made to addMark.

**Review by dylans — APPROVED** [review]

**Comment by dylans** [comment]
Ugh, I forgot to request a changeset for this before pressing the shiny merge button. @zhugexinxin could you please raise another PR with a changeset for this update? Thanks.

**Comment by zhugexinxin** [comment]
> Ugh, I forgot to request a changeset for this before pressing the shiny merge button. @zhugexinxin could you please raise another PR with a changeset for this update? Thanks.

ok, wait a moment

**Comment by zhugexinxin** [comment]
> Ugh, I forgot to request a changeset for this before pressing the shiny merge button. @zhugexinxin could you please raise another PR with a changeset for this update? Thanks.

@dylans 
sorry i want to confirm it, Do you want me to re-report another pr to overwrite the current one.
image

**Comment by zhugexinxin** [comment]
> Ugh, I forgot to request a changeset for this before pressing the shiny merge button. @zhugexinxin could you please raise another PR with a changeset for this update? Thanks.

@dylans 
I'm so sorry, I didn't understand what you meant at first, I deleted the branch when I saw the merge.
It seems that I can only roll back the previous merge and merge the current one, I am very sorry for this.
Or is there any other way?

**Pull request merged**
